docker/gremlin-server/Dockerfile.template

```diff
@@ -0,0 +1,27 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM tinkerpop:hadoop-HADOOP_VERSION
```

Comment:
I haven't really looked into this image, but couldn't we use the gremlin-server image that we also publish to Docker Hub instead as a base image?
This would have the nice advantage that we would use the image for testing that we are also publishing and we could also change the version to use an older image for the tests to test backwards compatibility.

Reply:
We could do that. The only challenge I see is copying the test data to the image.  Currently, it's copied from the working copy. What if you wanted to test 3.2.x server with a 3.4.x client? How do we access the test data from 3.2.x to copy it over?  How do we know if the test data is compatible?  Finally, what would this look like?  `./docker/gremlin-server.sh --version 3.2.9`?  Then without `--version` it uses the working copy?

Reply:
> What if you wanted to test 3.2.x server with a 3.4.x client? How do we access the test data from 3.2.x to copy it over?

You mean if the test data is version specific and therefore won't work with a 3.2.z server if we are on `master`?
I guess for that to work we would have to actually publish test images to Docker Hub that already contain the test data. But that should also be easier to do if we already use `tinkerpop/gremlin-server` here as the base image instead of building a completely new Gremlin Server image.

> Finally, what would this look like? ./docker/gremlin-server.sh --version 3.2.9? Then without --version it uses the working copy?

That sounds like a good idea to me, yes. It would make it possible to test different versions one after another by calling this script repeatedly with different versions.

But being able to test older versions was really only a possibility I saw here. I don't think that we should tackle that with this PR. If we can simplify testing for the image built from the current branch, then that's already a great improvement in my opinion.

Reply:
> I guess for that to work we would have to actually publish test images to Docker Hub that already contain the test data.

Just to reiterate something I mentioned earlier on the dev list - we should probably keep our docker image publishing restricted to public facing releases given some of the uncertainty on what the asf deems acceptable around this practice.

Reply:
I didn't want to say that we should do that now, but it would be the only option I see. Otherwise we would have to check out older commits to get test data for older versions which doesn't sound like a sane alternative to me.